src/linear_algebra/charpoly/basic.lean

```diff
@@ -67,12 +68,27 @@ begin
     split_ifs; simp [h], }
 end
 
+lemma charmatrix_reindex {m : Type v} [decidable_eq m] [fintype m] (e : n ≃ m)
+  (M : matrix n n R) : charmatrix (reindex e e M) = reindex e e (charmatrix M) :=
+begin
+  ext i j x,
+  by_cases h : i = j,
+  all_goals { simp [h] }
+end
+
 /--
 The characteristic polynomial of a matrix `M` is given by $\det (t I - M)$.
 -/
 def matrix.charpoly (M : matrix n n R) : polynomial R :=
 (charmatrix M).det
 
+lemma matrix.charpoly_of_reindex {m : Type v} [decidable_eq m] [fintype m] (e : n ≃ m)
```

Comment:
`of` is used in the naming system to mean `→`, so `charpoly_of_reindex` suggests a conclusion of the form `reindex _ _ _ = _ → charpoly _ = _`. We put the function names next to each other to mean "applied to".
```suggestion
lemma matrix.charpoly_reindex {m : Type v} [decidable_eq m] [fintype m] (e : n ≃ m)
```